vcd/resource_vcd_data_center_group.go

```diff
@@ -0,0 +1,452 @@
+			candidateVdcMap["remote_org"] = candidateVdc.RemoteOrg
+			candidateVdcMap["status"] = candidateVdc.Status
+			candidateVdcMap["org_name"] = candidateVdc.OrgRef.Name
+			candidateVdcMap["org_id"] = candidateVdc.OrgRef.ID
+			candidateVdcMap["site_name"] = candidateVdc.SiteRef.Name
+			candidateVdcMap["site_id"] = candidateVdc.SiteRef.ID
+			candidateVdcMap["vdc_name"] = candidateVdc.VdcRef.Name
+			candidateVdcMap["vdc_id"] = candidateVdc.VdcRef.ID
+
+			candidateVdcsSlice = append(candidateVdcsSlice, candidateVdcMap)
+		}
+	}
+
+	err := d.Set("participating_org_vdcs", schema.NewSet(schema.HashResource(participatingOrgVdcsResource), candidateVdcsSlice))
+	if err != nil {
+		return fmt.Errorf("[data center group read] could not set participating_org_vdcs block: %s", err)
+	}
+	return nil
+}
+
+func resourceVcdAlbDataCenterGroupDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf(errorRetrievingOrg, err)
+	}
+
+	vdcGroupToDelete, err := adminOrg.GetVdcGroupById(d.Id())
+	if err != nil {
+		return diag.Errorf("[data center group delete] : %s", err)
+	}
+
+	if vdcGroupToDelete.VdcGroup.DfwEnabled {
+		vdcGroupToDelete, err = vdcGroupToDelete.DisableDefaultPolicy()
+		if err != nil {
+			return diag.Errorf("error disabling default policy for data center group delete: %s", err)
+		}
+		vdcGroupToDelete, err = vdcGroupToDelete.DeActivateDfw()
+		if err != nil {
+			return diag.Errorf("error deactivating DFW for data center group delete: %s", err)
+		}
+	}
+
+	return diag.FromErr(vdcGroupToDelete.Delete())
+}
+
+func resourceDataCenterGroupImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 2 {
+		return nil, fmt.Errorf("resource name must be specified as org-name.data-center-group-name")
+	}
+	orgName, vdcGroupName := resourceURI[0], resourceURI[1]
+
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrg(orgName)
+	if err != nil {
+		return nil, fmt.Errorf("[data center group import] error retrieving org %s: %s", orgName, err)
+	}
+
+	vdcGroup, err := adminOrg.GetVdcGroupByName(vdcGroupName)
+	if err != nil {
+		return nil, fmt.Errorf("error importing data center group item: %s", err)
+	}
+
+	defaultValueStatus, err := getDefaultPolicyStatus(vdcGroup)
+	if err != nil {
+		return nil, fmt.Errorf("error importing data center group item: %s", err)
+	}
+
+	d.SetId(vdcGroup.VdcGroup.Id)
+	dSet(d, "org", orgName)
+	err = setVdcGroupConfigurationData(vdcGroup.VdcGroup, d, defaultValueStatus)
+	if err != nil {
+		return nil, fmt.Errorf("[data center group import] : %s", err)
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+// VdcGroupConfig is a minimal structure defining a VdcGroup in Organization
+type VdcGroupConfig struct {
```

Comment:
Can remain private I guess
```suggestion
type vdcGroupConfig struct {
```